server/platform/shared/filestore/s3store.go

```diff
@@ -56,6 +56,7 @@ type S3FileBackendNoBucketError struct{}
 const (
 	// This is not exported by minio. See: https://github.com/minio/minio-go/issues/1339
 	bucketNotFound = "NoSuchBucket"
+	invalidBucket  = "InvalidBucketName"
```

Comment:
With the new version, there is a distinction between no bucket, and invalid bucket name. So we have to handle that as well.

Reply:
That's a good distinction